gt/execution/run.hpp

```diff
@@ -92,8 +92,16 @@ result run(CommGrid &&comm_grid, Stepper &&stepper, real_t tmax, real_t dt,
       for (std::size_t k = 0; k < n.z; ++k)
         error = std::max(error, double(view(i, j, k) - expected(i, j, k)));
 
-  return {communication::global_max(comm_grid, error),
-          communication::global_max(comm_grid, time)};
+  return { error, time};
 }
 
-} // namespace execution
\ No newline at end of file
+template <class CommGrid, class Stepper, class Analytical>
```

Comment:
As there is only one use of `run` and two uses of `run_` and they have little difference, it feeld like we should remove `run` and move the additional stuff in there to the one usecase in `benchmark.cpp`